plugins/recorded_future/komand_recorded_future/util/api.py

```diff
@@ -27,8 +28,30 @@ def setup_custom_header(self):
 
     def _call_api(self, method: str, endpoint: str, params: dict = None, data: dict = None, json: dict = None):
         _url = self.base_url + endpoint
+        try:
+            response = requests.request(
+                url=_url, method=method, params=params, data=data, json=json, headers=self.headers
+            )
```

Comment:
If `MemoryError` is highly related to the `requests` module - did you try the response chunk? The `requests` package allows to enter boolean value to `stream` kwarg [https://requests.readthedocs.io/en/latest/api/#requests.request](). By means of that you are able to iterate through the chunks to pull all the content, limiting the memory usage [https://requests.readthedocs.io/en/latest/api/#requests.Response.iter_content](). For caching you can store full response content in file i.e. in `/var/cache` or something.

For example:
```python
response = requests.request(..., stream=True)
for chunk in response.iter_content(chunk_size=CHUNK_SIZE):  # imo 1024 would fit fine
    ...
```

You can create file on each action run with different `uuid` that exists only during the action execution. After retuning the response via Flask you can delete the file previously created, saving disk space.

This should be wrapped by handler

Reply:
Agreed but I think the issue is just passed to later in our code where we have to read and parse the file contents but we increase processing time. Due to the structure of the file we need to pass this in whole to convert XML to JSON.

I've updated to use streaming now, if this reduces any likelihood of a MemoryError it is probably worth it. Have refactored with more error handling between the steps.